bouter/embedded/__init__.py

```diff
@@ -6,90 +6,91 @@
+                segments,
+                continue_curvature=continue_curvature,
+                revert_pts=revert_pts,
+            )
+            self.behavior_log.loc[:, columns] = fixed_segments
+            self.behavior_log["missing_n"] = missing_n
 
-    @decorators.cache_results
-    def get_vigor(self, vigor_duration_s=0.05):
+        return self.behavior_log
+
+    @decorators.cache_results(cache_filename="behavior_log")
+    def compute_vigor(self, vigor_duration_s=0.05):
         """ Get vigor, the proxy of embedded fish forward velocity,
         a standard deviation calculated on a rolling window of tail curvature.
 
         :param vigor_duration: standard deviation window length in seconds
         :return:
         """
-        # TODO split in two methods so that it is set in the log df when
-        # loaded from cache (alternatively never write there)
-        if "vigor" in self.behavior_log.columns:
-            return self.behavior_log["vigor"]
-
         vigor_win = int(vigor_duration_s / self.behavior_dt)
         self.behavior_log["vigor"] = (
             self.behavior_log["tail_sum"]
             .interpolate()
             .rolling(vigor_win, center=True)
             .std()
         )
-        return self.behavior_log["vigor"]
-
-    def reconstruct_missing_segments(self, continue_curvature=0):
-        """ If the tail tip is not tracked throught the whole experiment
-        reconstruct the tail sum from the segments that are
-
-        """
-        utilities.fill_out_segments(
-            self.tail_points_matrix, continue_curvature=continue_curvature
-        )
+        return self.behavior_log
 
-    @decorators.cache_results
+    @decorators.cache_results()
     def get_bouts(self, vigor_threshold=0.1):
         """Extract bouts above threshold.
         :param vigor_threshold:
         :return:
         """
-        vigor = self.get_vigor()
+        # Make sure there's a vigor column:
+        self.compute_vigor()
         bouts, _ = utilities.extract_segments_above_threshold(
-            vigor.values, vigor_threshold
+            self.behavior_log["vigor"].values, vigor_threshold
         )
 
         return bouts
 
-    @decorators.cache_results
-    def get_bout_properties(self, bout_init_window_s=0.07):
+    @decorators.cache_results()
+    def get_bout_properties(self, directionality_duration=0.07):
         """Create dataframe with summary of bouts properties.
-        :param bout_init_window_s: Window defining initial part of
+        :param directionality_duration: Window defining initial part of
             the bout for the turning angle calculation, in seconds.
         :return:
         """
-        bout_init_window_pts = int(bout_init_window_s / self.behavior_dt)
+        bout_init_window_pts = int(directionality_duration / self.behavior_dt)
         tail_sum = self.behavior_log["tail_sum"].values
-        vigor = self.get_vigor().values
+        vigor = self.compute_vigor().values
```

Comment:
not sure it's correct if compute vigor returns the whole dataframe

Reply:
No it was a bug